app/controllers/OwnBrandsController.scala

```diff
@@ -16,47 +16,72 @@
 
 package controllers
 
+import config.FrontendAppConfig
 import controllers.actions._
+import errors.NoPendingReturnForGivenPeriod
 import forms.OwnBrandsFormProvider
 import handlers.ErrorHandler
-import models.{Mode, UserAnswers}
+import models.requests.OptionalDataRequest
+import models.{CheckMode, Mode, NormalMode, UserAnswers}
 import navigation.Navigator
+import orchestrators.ReturnsOrchestrator
 import pages.{BrandsPackagedAtOwnSitesPage, OwnBrandsPage}
+import play.api.data.Form
 import play.api.i18n.MessagesApi
-import play.api.mvc.{Action, AnyContent, MessagesControllerComponents}
+import play.api.mvc.{Action, AnyContent, MessagesControllerComponents, Result}
 import repositories.SessionRepository
+import uk.gov.hmrc.http.HeaderCarrier
 import utilitlies.GenericLogger
 import views.html.OwnBrandsView
 
 import javax.inject.Inject
 import scala.concurrent.{ExecutionContext, Future}
 
-class OwnBrandsController @Inject()(
+class OwnBrandsController @Inject()(returnsOrchestrator: ReturnsOrchestrator,
                                      override val messagesApi: MessagesApi,
                                      val sessionRepository: SessionRepository,
                                      val navigator: Navigator,
                                      val errorHandler: ErrorHandler,
                                      val genericLogger: GenericLogger,
                                      identify: IdentifierAction,
                                      getData: DataRetrievalAction,
+//                                     requireData: DataRequiredAction,
+//                                     checkReturnSubmission: CheckingSubmissionAction,
                                      formProvider: OwnBrandsFormProvider,
                                      val controllerComponents: MessagesControllerComponents,
-                                     view: OwnBrandsView
+                                     view: OwnBrandsView,
+                                     config: FrontendAppConfig
                                    )(implicit ec: ExecutionContext) extends ControllerHelper {
 
   private val form = formProvider()
+ //Todo use this function once ATs route to the ReturnsController from dashboard and use required data and checkSubmission actions
```

Comment:
can all of this be a story please, easiest way, print screen the code, and explain and put a timeline as to when it needs to be done, also mention theres todos in x y and z places assosicated with it